ecosystem/typescript/sdk/src/token_client.ts

```diff
@@ -570,7 +571,21 @@ export class TokenClient {
 
     // We know the response will be a struct containing TokenData, hence the
     // implicit cast.
-    return this.aptosClient.getTableItem(handle, getTokenTableItemRequest);
+    return this.aptosClient
+      .getTableItem(handle, getTokenTableItemRequest)
+      .then(
+        (rawTokenData) =>
+          new TokenData(
+            rawTokenData.collection,
+            rawTokenData.description,
+            rawTokenData.name,
+            rawTokenData.maximum,
+            rawTokenData.supply,
+            rawTokenData.uri,
+            rawTokenData.default_properties,
+            rawTokenData.mutability_config,
+          ),
+      );
```

Comment:
We barely use the promise chaining. To be consistent, let's use the `await async`. This can be written as 
```
const rawTokenData = await this.aptosClient.getTableItem(handle, getTokenTableItemRequest);
return new TokenData(
            rawTokenData.collection,
            rawTokenData.description,
            rawTokenData.name,
            rawTokenData.maximum,
            rawTokenData.supply,
            rawTokenData.uri,
            rawTokenData.default_properties,
            rawTokenData.mutability_config,
          );
```

Reply:
updated